benchmark/benchmark.hpp

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+/*
+ * This file defines a benchmarking framework for evaluating memory allocation
```

Comment:
Kinda looks like wall of text... perhaps, please make it more concise.

I really like the flow description and "extension" section, though